test/test_common/utility.h

```diff
@@ -199,44 +199,60 @@ class TestUtility {
   static Stats::GaugeSharedPtr findGauge(Stats::Store& store, const std::string& name);
 
   /**
-   * Wait till Counter value is equal to the passed ion value.
+   * Wait for a counter to == a given value.
    * @param store supplies the stats store.
    * @param name supplies the name of the counter to wait for.
    * @param value supplies the value of the counter.
    * @param time_system the time system to use for waiting.
+   * @param timeout the maximum time to wait before timing out.
+   * @return AssertionSuccess() if the counter was == to the value within the timeout, else
+   * AssertionFailure().
    */
-  static void waitForCounterEq(Stats::Store& store, const std::string& name, uint64_t value,
-                               Event::TestTimeSystem& time_system);
+  static AssertionResult waitForCounterEq(Stats::Store& store, const std::string& name,
+                                          uint64_t value, Event::TestTimeSystem& time_system,
+                                          std::chrono::milliseconds timeout = DefaultTimeout);
 
   /**
    * Wait for a counter to >= a given value.
    * @param store supplies the stats store.
    * @param name counter name.
    * @param value target value.
    * @param time_system the time system to use for waiting.
+   * @param timeout the maximum time to wait before timing out.
+   * @return AssertionSuccess() if the counter was == to the value within the timeout, else
+   * AssertionFailure().
    */
-  static void waitForCounterGe(Stats::Store& store, const std::string& name, uint64_t value,
-                               Event::TestTimeSystem& time_system);
+  static AssertionResult waitForCounterGe(Stats::Store& store, const std::string& name,
+                                          uint64_t value, Event::TestTimeSystem& time_system,
+                                          std::chrono::milliseconds timeout = DefaultTimeout);
 
   /**
    * Wait for a gauge to >= a given value.
    * @param store supplies the stats store.
    * @param name gauge name.
    * @param value target value.
    * @param time_system the time system to use for waiting.
+   * @param timeout the maximum time to wait before timing out.
```

Comment:
Is there a way to prevent this from using real time? For tests I think we can only safely do this with simulated time. Tests already have a bounded lifetime, so we can rely on that for the real time counter wait timeout.

Reply:
The main reason for doing this was that the real timeout is something like 5 minutes so debugging is kind of difficult when you have to wait so long to see the logs, etc.  I can check to see if I can just condition on whether time_system_ is simulated time or not, but not sure how useful a timeout in simulated time would be.  What do you think?

Reply:
per offline discussion sam's changing the timeout to an optional, so users will fail on timeout only when a timeout param is passed